src/components/Hosts/UpdateStatusModal.tsx

```diff
@@ -0,0 +1,216 @@
+  visible,
+  dataCy,
+  hostIds,
+  closeModal,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+
+  const [status, setHostStatus] = useState<HostStatus | null>(null);
+
+  const [notes, setNotesValue] = useState<string>("");
+
+  const resetForm = () => {
+    setHostStatus(null);
+    setNotesValue("");
+  };
+
+  // UPDATE HOST STATUS MUTATION
+  const [updateHostStatus, { loading: loadingUpdateHostStatus }] = useMutation<
+    UpdateHostStatusMutation,
+    UpdateHostStatusMutationVariables
+  >(UPDATE_HOST_STATUS, {
+    onCompleted({ updateHostStatus: numberOfHostsUpdated }) {
+      closeModal();
+      dispatchBanner.successBanner(
+        `Status was changed to ${status} for ${numberOfHostsUpdated} host${
+          numberOfHostsUpdated === 1 ? "" : "s"
+        }`
+      );
+      resetForm();
+    },
+    onError(error) {
+      closeModal();
+      dispatchBanner.errorBanner(
+        `There was an error updating hosts status: ${error}`
+      );
+    },
+  });
+
+  const onClickUpdate = () => {
+    updateHostStatus({ variables: { hostIds, status, notes } });
+  };
+
+  const onClickCancel = () => {
+    closeModal();
+    resetForm();
+  };
+
+  return (
+    <Modal
+      data-cy={dataCy}
+      visible={visible}
+      onCancel={onClickCancel}
+      title="Update Host Status"
+      footer={
+        <Buttons>
+          <ButtonWrapper>
+            <Button dataCy="modal-cancel-button" onClick={onClickCancel}>
+              Cancel
+            </Button>
+          </ButtonWrapper>
+          <Button
+            dataCy="modal-update-button"
+            variant="primary"
+            disabled={!status}
+            loading={loadingUpdateHostStatus}
+            onClick={onClickUpdate}
+          >
+            Update
+          </Button>
+        </Buttons>
+      }
+    >
+      <Body weight="medium">Host Status</Body>
+      <StyledSelect
+        data-cy="host-status-select"
+        value={status}
+        onChange={(s) => setHostStatus(s as HostStatus)}
+      >
+        {hostStatuses.map(({ title, value, key }) => (
+          <Option key={key} value={value} data-cy={`${value}-option`}>
+            {title}
+          </Option>
+        ))}
+      </StyledSelect>
+      <Body weight="medium">Add Notes</Body>
+      <StyledTextArea
+        data-cy="host-status-notes"
+        value={notes}
+        autoSize={{ minRows: 4, maxRows: 6 }}
+        onChange={(e) => setNotesValue(e.target.value)}
+      />
+    </Modal>
+  );
+};
+
+// STYLES
+const Buttons = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+`;
+const ButtonWrapper = styled.div`
+  margin-right: 16px;
+`;
+const StyledSelect = styled(Select)`
+  width: 100%;
+  margin-bottom: 24px;
+`;
+const StyledTextArea = styled(TextArea)`
+  resize: none;
+`;
+
+// HOSTS STATUSES DATA FOR SELECT COMPONENT
+interface Status {
+  title: keyof typeof HostStatus;
+  value: HostStatus;
+  key: HostStatus;
+}
+
+const hostStatuses: Status[] = [
+  {
```

Comment:
Some of these statuses dont make sense in this context. This modal should update the status of the host by performing some type of action such as start, stop, terminate. The statuses are the result of that action.
 It shouldn't be possible for a user to directly give a host a stopping state they instead must perform a stop action and if successful the state of the host should become stopping, then stopped. 

You can reference some of the current actions here. https://github.com/evergreen-ci/evergreen/blob/de3debcf884440844f6a3d71b44b1ec36496d9a4/service/spawn.go#L32-L34

Reply:
great! thanks